requests/src/main/scala/weco/api/requests/RequestsApi.scala

```diff
@@ -23,33 +24,36 @@ class RequestsApi(
 ) extends CreateRequest
     with LookupPendingRequests {
   val routes: Route = concat(
-    RequestsApi.withUserId("users" / Segment / "item-requests") {
-      userIdentifier: SierraPatronNumber =>
-        post {
-          entity(as[ItemRequest]) {
-            itemRequest: ItemRequest =>
-              // TODO: We get the work ID as part of the item request, although right now
-              // it's only for future-proofing, in case it's useful later.
-              // Should we query based on the work ID?
-              Try { CanonicalId(itemRequest.itemId) } match {
-                case Success(itemId) =>
-                  withFuture {
-                    createRequest(
-                      itemId = itemId,
-                      patronNumber = userIdentifier
-                    )
-                  }
+    RequestsApi
+      .withUserId("users" / Segment / "item-requests") {
+        userIdentifier: SierraPatronNumber =>
+          post {
+            entity(as[ItemRequest]) {
+              itemRequest: ItemRequest =>
+                // TODO: We get the work ID as part of the item request, although right now
+                // it's only for future-proofing, in case it's useful later.
+                // Should we query based on the work ID?
+                Try { CanonicalId(itemRequest.itemId) } match {
+                  case Success(itemId) =>
+                    withFuture {
+                      createRequest(
+                        itemId = itemId,
+                        neededBy =
+                          itemRequest.neededBy.getOrElse(LocalDate.now()),
```

Comment:
Do you mind putting another TODO here? It reads as a bit confusing unless you know why we've done it